src/PerfView/HeapView/Toolbox.cs

```diff
@@ -787,19 +787,24 @@ internal static List<IProcess> GetClrProcesses(this TraceLog traceLog, out bool
                             {
                                 clrVersion = m.ModuleFile.FilePath;
 
-                                int pos = clrVersion.LastIndexOf('\\');
-
-                                if (pos > 0)
+                                // Only extract version from path for clr.dll, for other modules use full path
+                                if (name.Equals("clr", StringComparison.OrdinalIgnoreCase))
```

Comment:
CONSIDER: This is all more easily readable with Path.GetFileName(Path.GetDirectoryName(path)) with maybe some sanity checks around the input.